addon/index.js

```diff
@@ -451,7 +450,12 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
 
       if (validation === true || isSingleValidationArray) {
         this._deleteKey(ERRORS, key);
-        set(changes, key, value);
+
+        if (oldValue !== value) {
```

Comment:
I'm guessing this is not a good enough way to compare equality. It's possible that both of these are objects (as I get using ember-ted-select), so this would not properly detect equality there.